utils/incremental_refresh/util.sql

```diff
@@ -1,185 +1,140 @@
+                _timescaledb_additional.cagg_time_bucket(mat_hypertable_id, range.start) AS global_start,
+                _timescaledb_additional.cagg_time_bucket(mat_hypertable_id, range.end) + (bf.bucket_width::interval + '1 millisecond'::interval) AS global_end
+            FROM _timescaledb_catalog.continuous_agg cagg
+            JOIN _timescaledb_catalog.continuous_aggs_bucket_function bf USING (mat_hypertable_id)
+            JOIN _timescaledb_catalog.hypertable ht ON (ht.id = cagg.raw_hypertable_id)
+            JOIN _osm_catalog.table_map tm ON (tm.hypertable_name = ht.table_name AND tm.hypertable_schema = ht.schema_name)
+            JOIN (
+                -- the time window of tiered data
+                SELECT
+                    osm_table_id,
+                    _osm_internal.dimension_pg_usec_to_timestamp(min(range_start)) as start,
+                    _osm_internal.dimension_pg_usec_to_timestamp(max(range_end)) as end
+                FROM _osm_catalog.chunk_map
+                JOIN _osm_catalog.chunk_object_map USING (chunk_id)
+                GROUP BY osm_table_id
+            ) AS range USING (osm_table_id)
+        )
         SELECT
-            cagg.mat_hypertable_id,
-            ht.schema_name,
-            ht.table_name,
-            (
-                SELECT column_type as dim_type
-                FROM _timescaledb_catalog.dimension d
-                WHERE d.hypertable_id = ht.id
-                ORDER BY d.id ASC LIMIT 1
-            ) as dim_type,
+            mat_hypertable_id,
+            dim_type,
             user_view_schema,
             user_view_name,
-            range.start as range_start,
-            range.end as range_end
-        FROM _timescaledb_catalog.continuous_agg cagg
-        JOIN _timescaledb_catalog.hypertable ht ON (ht.id = cagg.raw_hypertable_id)
-        JOIN _osm_catalog.table_map tm ON (tm.hypertable_name = ht.table_name AND tm.hypertable_schema = ht.schema_name)
-        JOIN (
-            -- the time window of tiered data
-            SELECT
-                osm_table_id,
-                _osm_internal.dimension_pg_usec_to_timestamp(min(range_start)) as start,
-                _osm_internal.dimension_pg_usec_to_timestamp(max(range_end)) as end
-            FROM _osm_catalog.chunk_map
-            JOIN _osm_catalog.chunk_object_map USING (chunk_id)
-            GROUP BY osm_table_id
-        ) as range USING (osm_table_id)
+            global_start,
+            global_end,
+            start,
+            start + (bucket_width * 5) AS end,
+            (extract(epoch from start) * 1000000)::bigint AS invalidation_start,
+            (extract(epoch from (start + (bucket_width * nbuckets))) * 1000000)::bigint AS invalidation_end
+        FROM
+            ranges,
+            -- Split ranges with 5 times the bucket width
+            LATERAL generate_series(ranges.global_start, ranges.global_end, (bucket_width * nbuckets)) AS start
```

Comment:
I've been looking through the changes, and I think one thing is missing from my original code. Molex has this weird data points in year 1917 or so (probably inserted by mistake). And apparently they want to keep it that way. If we generate ranges with this query it would create all the ranges between 1917 and 2024, while they only have few data points in pre-2020. In my original code I had this join to only generate the ranges that intersect with existing tiered chunks:
```sql
        FROM timescaledb_osm.tiered_chunks ch
        JOIN _timescaledb_additional.generate_increments(start_t, end_t, increment_size) AS i
            ON tstzrange(i.incr_start, i.incr_end, '[)') && tstzrange(ch.range_start, ch.range_end, '[)')
```